app/main.py

```diff
@@ -1,2 +1,53 @@
+from typing import Any, Hashable
+
+
 class Dictionary:
-    pass
+    # noinspection PyCompatibility
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.load_factor = 2 / 3
+        self.size = 0
+        self.hash_table: list = [None] * 8
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        if self.size >= self.capacity * self.load_factor:
+            self.resize()
+        if self.hash_table[self.count_index(key)] is None:
```

Comment:
you can use `if not` instead of `is None`

you can save `self.count_index(key_hash) `in a variable